src/appliance-onboarding-script/appliance_setup/run.py

```diff
@@ -145,6 +143,11 @@ def _populate_default_values_of_optional_fields_in_config(config):
             raise InvalidRegion(f"This feature is only available in these regions: {Constant.VALID_LOCATIONS}")
 
     if operation == 'onboard':
+        arc_addon_creator = ArcAddonCreator()
+        onboard_state =  arc_addon_creator.check_if_arc_onboarded_on_cloud(_customer_details.customer_resource)
```

Comment:
Does it add value to include this method in the creator class?
We don't really create the ArcAddOn here. Also while registering, another object of the same type gets created in the method **register_with_private_cloud**

Reply:
avs_orchestrator class might bloat up to hold everything then. 
Also, this is not related to orchestrator in my understanding. 

It would make sense to have all arc related logic in common place in my opinion. However, it might not best fit the creater / remover pattern we have already.

If we follow the pattern followed already, validations generally go to the pre-processing function in the specific processor. However, we do not have a processor for arc addon, at least yet. 

W

Reply:
Do we anticipate more changes that might fit into the avs_orchestrator class?
I think whatever you said makes sense, but if this is high effort, should we take it separately or on need basis?

Reply:
@uthsab @msfttusharjuneja. created a new file for arc_add_on retriever and placed it in retriever/arc_add_on. This would be a more consistent logic as per existing pattern. LMK your thoughts